### PR TITLE
Send verification error to Caboose

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Send verification error to Caboose.

### CODE REVIEW

**Comment by aarshkshah1992** [comment]
Caboose PR at https://github.com/filecoin-saturn/caboose/pull/141.

File: lib/gateway_traversal.go [hunk 1]
```diff
@@ -94,11 +95,15 @@ func carToLinearBlockGetter(ctx context.Context, reader io.Reader, metrics *Grap
...
-				return nil, fmt.Errorf("unexpected block received: expected %s, got %s", c, blkRead.block.Cid())
+				return nil, caboose.ErrInvalidResponse{
+					Message: fmt.Sprintf("unexpected block received: expected %s, got %s", c, blkRead.block.Cid()),
+				}
 			}
 			return blkRead.block, nil
 		}
-		return nil, fmt.Errorf("received a nil block with no error")
+		return nil, caboose.ErrInvalidResponse{
+			Message: fmt.Sprintf("received a nil block with no error"),
```
Comment (willscott) [inline lib/gateway_traversal.go:105]: make this a constant error?

**Comment by aarshkshah1992** [comment]
Failing to build because of Error: ../../../go/pkg/mod/go.opentelemetry.io/otel/exporters/otlp/otlptrace@v1.14.0/internal/otlpconfig/options.go:100:62: undefined: internal.GetUserAgentHeader.

**Comment by willscott** [comment]
@aarshkshah1992  - pull in the `staging` branch - it's got the fix for dependencies / current otel versions

**Comment by aarshkshah1992** [comment]
@willscott You mean the Bifrost staging version or Caboose staging version ?

**Comment by willscott** [comment]
i mean https://github.com/ipfs/bifrost-gateway/commit/06077bc95feeb38f5eb26e9bac6ca896cc5959f6

**Review by willscott — APPROVED** [review]

**Pull request merged**